consistency-checker/src/test/java/com/redhat/lightblue/migrator/consistency/MigrationJobTest.java

```diff
@@ -139,6 +216,147 @@ public void testDocumentsConsistent_SourceAndDestinationNotConsistent(){
+    }
+
+    /**
+     * Ensures that a more complex json document with a top level array, with a deep invalid field but where that field is excluded.
+     * Should pass.
+     */
+    @Test
+    public void testDocumentsConsistent_ComplexJson_WithTopLevelArray_Pass() throws Exception{
+        ObjectMapper mapper = new ObjectMapper();
+
+        JsonNode source = mapper.readTree(
+                "[{\"field1\": \"value1\", \"something\": \"x\", \"array\": [{\"field2\": \"value2\",\"something\": \"y\"}]}]");
+
+        JsonNode dest = mapper.readTree(
+                "[{\"field1\": \"value1\", \"something\": \"x\", \"array\": [{\"field2\": \"value2\",\"something\": \"z\"}]}]");
+
+        MigrationConfiguration jobConfiguration = new MigrationConfiguration();
+        jobConfiguration.setComparisonExclusionPaths(Arrays.asList("array.something"));
+        migrationJob.setJobConfiguration(jobConfiguration);
+
+        assertTrue(migrationJob.documentsConsistent(source, dest));
+    }
+
+    /**
+     * Ensures that a more complex json document with a top level array, with a deep invalid field.
+     * Should fail.
+     */
+    @Test
+    public void testDocumentsConsistent_ComplexJson_WithTopLevelArray_Fail() throws Exception{
+        ObjectMapper mapper = new ObjectMapper();
+
+        JsonNode source = mapper.readTree(
+                "[{\"field1\": \"value1\", \"something\": \"x\", \"array\": [{\"field2\": \"value2\",\"something\": \"y\"}]}]");
+
+        JsonNode dest = mapper.readTree(
+                "[{\"field1\": \"value1\", \"something\": \"x\", \"array\": [{\"field2\": \"value2\",\"something\": \"z\"}]}]");
+
+        assertFalse(migrationJob.documentsConsistent(source, dest));
+    }
+
+    /**
+     * Exclude "something" instead of "array.something" and verify documents are not consistent.
+     * Should fail.
+     */
+    @Test
+    public void testDocumentsConsistent_ComplexJson_WithTopLevelArray_ExcludeNotOnField_Fail() throws Exception{
+        ObjectMapper mapper = new ObjectMapper();
+
+        JsonNode source = mapper.readTree(
+                "[{\"field1\": \"value1\", \"something\": \"x\", \"array\": [{\"field2\": \"value2\",\"something\": \"y\"}]}]");
+
+        JsonNode dest = mapper.readTree(
+                "[{\"field1\": \"value1\", \"something\": \"x\", \"array\": [{\"field2\": \"value2\",\"something\": \"z\"}]}]");
+
+        MigrationConfiguration jobConfiguration = new MigrationConfiguration();
+        jobConfiguration.setComparisonExclusionPaths(Arrays.asList("something"));
+        migrationJob.setJobConfiguration(jobConfiguration);
+
+        assertFalse(migrationJob.documentsConsistent(source, dest));
+    }
+
+    /**
+     * Have a different value for "something" at the top level, verify docs are not consistent
+     * Should fail.
+     */
+    @Test
+    public void testDocumentsConsistent_ComplexJson_WithTopLevelMismatch_fail() throws Exception{
```

Comment:
I think everything is tested at a low level, but it is good to have some complex json tests also.
Changes made.